src/GraphQL/Types/Scalars/HalfGraphType.cs

```diff
@@ -0,0 +1,68 @@
+#if NET5_0_OR_GREATER
+
+using System.Numerics;
+using GraphQLParser;
+using GraphQLParser.AST;
+
+namespace GraphQL.Types;
+
+/// <summary>
+/// The Half scalar graph type represents an IEEE 754 half-precision floating point value.
+/// </summary>
+public class HalfGraphType : ScalarGraphType
+{
+    /// <inheritdoc/>
+    public override object? ParseLiteral(GraphQLValue value) => value switch
+    {
+        GraphQLIntValue x => Parse(value, x.Value),
+        GraphQLFloatValue x => Parse(value, x.Value),
+        GraphQLNullValue _ => null,
+        _ => ThrowLiteralConversionError(value)
+    };
+
+    /// <inheritdoc/>
+    public override bool CanParseLiteral(GraphQLValue value) => value switch
+    {
+        GraphQLIntValue x => TryParse(x.Value),
+        GraphQLFloatValue x => TryParse(x.Value),
+        GraphQLNullValue _ => true,
+        _ => false
+    };
+
+    private object Parse(GraphQLValue value, ROM rom)
+    {
+        var parsed = Half.Parse(rom);
+        return Half.IsInfinity(parsed) ? ThrowLiteralConversionError(value) : parsed;
+    }
+
+    private bool TryParse(ROM rom)
+    {
+        return Half.TryParse(rom, out var parsed) && !Half.IsInfinity(parsed);
+    }
+
+    private object NotInfinity(Half value)
+    {
+        return Half.IsInfinity(value) ? ThrowValueConversionError(value) : value;
+    }
+
+    /// <inheritdoc/>
+    public override object? ParseValue(object? value) => value switch
+    {
+        double db => NotInfinity(checked((Half)db)),
+        int i => NotInfinity(checked((Half)i)),
```

Comment:
```suggestion
        int i => checked((Half)i),
```
Integer conversions should throw if out of range

Reply:
![изображение](https://user-images.githubusercontent.com/21261007/202756844-a0db4b02-02da-47c7-81bd-096ca4f059af.png)

I return `NonInfinity` back.

Reply:
Ok